### DISCUSSION TITLE
Intel compilers for MET becoming deprecated

### DISCUSSION THREAD

**Discussion by HathewayWill** [dtcenter/METplus#1957] [Installation] [answered]
I wanted to share this error I got with when trying to set up intel with met and met plus.  I thought you should know about this.  

icc: remark #10441: The Intel(R) C++ Compiler Classic (ICC) is deprecated and will be removed from product release in the second half of 2023. The Intel(R) oneAPI DPC++/C++ Compiler (ICX) is the recommended compiler moving forward. Please transition to use this compiler. Use '-diag-disable=10441' to disable this message.

MET 5.0 may need to add the new compilers as options. 

CC= icx
CXX=icpx
FC/F77/F90= ifx
MPIFC/F77/F90= MPIIFORT -FC=IFX
MPIFC= MPIIFORT -FC=IFX
MPICC= MPIIFORT -CC=ICX
MPICXX= MPIIFORT -CXX=ICPX

**Comment by HathewayWill** [comment, marked as answer]
export CC=icx
export CXX=icpx
export FC=ifx
export F77=ifx
export F90=ifx
export MPIFC='mpiifort -fc=ifx'
export MPIF77='mpiifort -fc=ifx'
export MPIF90='mpiifort -fc=ifx'
export MPICC='mpiicc -cc=icx'
export MPICXX='mpiicpc -cxx=icpx'

**Comment by jprestop** [comment]
Thanks, @HathewayWill.  I do see notice of that on the Intel webpage here:
https://www.intel.com/content/www/us/en/developer/articles/release-notes/oneapi-c-compiler-release-notes.html

Thank you for this information.  I suspect it will be useful for our users and also useful in updating the compile_MET_all.sh script once Intel creates their new release and deprecates icc and we have access to the new Intel compiler for testing.

I'll go ahead and mark this as answered and lock this Discussion.
